sds_data_manager/lambda_code/IAlirtCode/ialirt_data_query_api.py

```diff
@@ -0,0 +1,245 @@
+        lambda function to process
+    context : LambdaContext
+        This object provides methods and properties that provide
+        information about the invocation, function,
+        and runtime environment.
+
+    """
+    params = event.get("queryStringParameters") or {}
+
+    # --- Determine key condition ---
+    allowed_params = {
+        "instrument",
+        "time_utc_start",
+        "time_utc_end",
+        "met_in_utc_start",  # for backward compatibility
+        "met_in_utc_end",  # for backward compatibility
+    }
+
+    # Ensure allowed parameters
+    unexpected = set(params) - allowed_params
+    if unexpected:
+        return _error(400, f"Unexpected parameters: {', '.join(unexpected)}")
+
+    if not params.get("instrument"):
+        meta_instrument = "all"
+        meta_type = "science"
+    elif params["instrument"] == "spice":
+        meta_instrument = "spice"
+        meta_type = "spice"
+    elif params["instrument"].endswith("hk"):
+        meta_instrument = params["instrument"]
+        meta_type = "hk"
+    else:
+        meta_instrument = params["instrument"]
+        meta_type = "science"
+
+    # Get instrument or default to all.
+    requested_instrument = params.get("instrument")
+    instruments = (
+        [requested_instrument]
+        if requested_instrument
+        else ["hit", "mag", "codice_lo", "codice_hi", "swapi", "swe"]
+    )
+
+    items = []
+    query_time_total = 0
+
+    for instrument in instruments:
+        key_expr = Key("instrument").eq(instrument)
+        query_kwargs = {"KeyConditionExpression": key_expr}
+
+        if any(
+            param in params
+            for param in (
+                "time_utc_start",
+                "time_utc_end",
+                "met_in_utc_start",
+                "met_in_utc_end",
+            )
+        ):
+            result = apply_time_filters(params, query_kwargs)
+            if result is None:
+                return _error(400, "Invalid time format: expected YYYY-MM-DDTHH:MM:SS")
+            query_kwargs, range_start, range_end = result
+            # Checks if the max time range is exceeded.
+            if abs(
+                datetime.fromisoformat(range_end) - datetime.fromisoformat(range_start)
+            ) > timedelta(days=1):
+                return _error(400, "Start and end time cannot exceed 1 day apart.")
+        else:
+            # Get latest 1 hour if not specified.
+            logger.info(
+                "No time range specified, defaulting to last 1 hour for instrument: %s",
+                instrument,
+            )
+            now = datetime.now(timezone.utc)
+            one_hour_ago = now - timedelta(hours=1)
+            query_kwargs["KeyConditionExpression"] &= Key("time_utc").between(
+                one_hour_ago.isoformat(), now.isoformat()
+            )
+            range_start = one_hour_ago.isoformat()
+            range_end = now.isoformat()
```

Comment:
This too I think can go into the final clause of apply_time_filters to say if there was no start/end use the default code you have here.